src/AutorestSwift/JsonRpc/AutorestPlugin.swift

```diff
@@ -165,22 +165,16 @@ class AutorestPlugin {
         guard let codeModel = response.asString else {
             SharedLogger.fail("Unable to retrieve code model from Autorest.")
         }
-        // FIXME: Wonky workaround to pass in package name
-        let packageNameRequest: RPCObject = .list([.string(sessionId), .string("package-name")])
-        let getValFuture = client.call(method: "GetValue", params: packageNameRequest)
-        getValFuture.whenSuccess { result in
-            var packageName: String?
-            switch result {
-            case let .success(response):
-                packageName = response.asString
-            case .failure:
-                packageName = nil
-            }
-            let manager = Manager(withString: codeModel, packageName: packageName)
+        Manager.shared.configure(input: codeModel, client: client, sessionId: sessionId) {
+            let encoder = JSONEncoder()
+            encoder.outputFormatting = .prettyPrinted
+            let encoded = try! encoder.encode(Manager.shared.config)
+            let arguments = String(data: encoded, encoding: .utf8)!
+            SharedLogger.info("Autorest Configuration: \(arguments)")
             do {
-                try manager.run()
+                try Manager.shared.run()
 
-                guard let packageUrl = manager.packageUrl else {
+                guard let packageUrl = Manager.shared.packageUrl else {
```

Comment:
Why we do this check (check if packageUrl is nil) here (in line 177) after manager is run()? shouldn't we check this before Manager.run?

Reply:
I'll have to check, but I think packageUrl is set during `run`.

Reply:
Yes, this is set during `run()`.